### PR TITLE
[Pipeline] Add zero shot audio classificatoin pipeline

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Add the `zero_shot_audio_classification_pipeline`  for the `CLAP` models. See #21370

### CODE REVIEW

File: src/transformers/pipelines/zero_shot_audio_classification.py
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:35]:
```suggestion
    >>> classifier = pipeline(task="zero-shot-audio-classification", model="laion-ai/clap-htsat-tiny")
```
it's `htsat` no?

File: src/transformers/pipelines/zero_shot_audio_classification.py
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:40]:
```suggestion
    ... )
    [{'score': 0.999727189540863, 'label': 'Sound of a dog'}, {'score': 0.0002727957325987518, 'label': 'Sound of vaccum cleaner'}]
```
Need to check the score otherwise the doc-test (if any) won't probably pass?

File: src/transformers/pipelines/zero_shot_audio_classification.py
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:56]:
```suggestion
```

**Review by younesbelkada — APPROVED** [review]
Thanks for your great work! Left few nits - 
let's wait for @Narsil 's approval before merging

File: src/transformers/pipelines/zero_shot_audio_classification.py
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:36]:
```suggestion
    >>> pipeline_output = classifier(
```
Something like that combined with the suggestion below should do the trick too

**Review by Narsil — APPROVED** [review]

File: src/transformers/pipelines/base.py
Comment (ArthurZucker) [inline src/transformers/pipelines/base.py:80]: Failing test might be related ot this! Will investigate

Comment (ArthurZucker) [reply src/transformers/pipelines/base.py:80]: Indeed, fixing this then will be good to go! 🔥

**Comment by Narsil** [comment]
LGTM !

File: src/transformers/pipelines/zero_shot_audio_classification.py
Comment (sgugger) [inline src/transformers/pipelines/zero_shot_audio_classification.py:57]: Looks like this fits in one line.

File: src/transformers/pipelines/zero_shot_audio_classification.py
Comment (sgugger) [inline src/transformers/pipelines/zero_shot_audio_classification.py:60]: The args should be below the description.

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Pull request merged**
